### PR TITLE
deduplicate the 'isSomeFunction' equivalent

### PR DESCRIPTION
**Pull request by UplinkCoder**
as per title

### CODE REVIEW

**Comment by JinShil** [comment]
FYI:  This is a resubmission of #6535 because the Phantom Zone thing didn't work out well.  We couldn't re-open the PR after @UplinkCoder  made changes.

File: src/dmd/traits.d [hunk 1]
```diff
@@ -443,6 +444,51 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
...
+        o = an AST node to check for a `TypeFunction`
+        fdp = optional pointer to a function declararion, to be set
+          if `o` is a function declarartion.
+
+     Returns:
+        a type node if `o` is a declaration of
+            a delegate, function, function-pointer
+          or a variable of the former.  Otherwise, `null`.
+    */
+    static TypeFunction toTypeFunction(RootObject o, FuncDeclaration* fdp = null)
```
[the remark this answers is not in the document]
Comment (JinShil) [reply src/dmd/traits.d]: UPDATE 2: OK, I just realized the outer function context is not required, so `static` seems to make sense.  I'm good, now.  Sorry to beat this horse so much.

**Review by JinShil — APPROVED** [review]
I'm going to leave a short window of opportunity for others to add their review before I merge.

**Comment by WalterBright** [comment]
My review is here https://github.com/dlang/dmd/pull/7754 :-)

**Pull request merged**